ingestion/src/metadata/ingestion/source/kafka.py

```diff
@@ -49,50 +55,51 @@ def dropped(self, topic: str) -> None:
         self.filtered.append(topic)
 
 
-class KafkaSourceConfig(ConfigModel):
-    bootstrap_servers: str = "localhost:9092"
-    schema_registry_url: str = "http://localhost:8081"
-    consumer_config: dict = {}
-    service_name: str
-    service_type: str = MessagingServiceType.Kafka.value
-    filter_pattern: IncludeFilterPattern = IncludeFilterPattern.allow_all()
-
-
 @dataclass
 class KafkaSource(Source[CreateTopicRequest]):
-    config: KafkaSourceConfig
+    config: WorkflowSource
     admin_client: AdminClient
     report: KafkaSourceStatus
 
     def __init__(
         self,
-        config: KafkaSourceConfig,
+        config: WorkflowSource,
         metadata_config: OpenMetadataConnection,
     ):
         super().__init__()
         self.config = config
+        self.source_config = self.config.sourceConfig.config
+        self.service_connection = self.config.serviceConnection.__root__.config
         self.metadata_config = metadata_config
         self.status = KafkaSourceStatus()
+        self.kafka_service_config = {
+            "bootstrapServers": self.service_connection.bootstrapServers,
+            "schemaRegistryURL": self.service_connection.schemaRegistryURL,
+        }
         self.service = get_messaging_service_or_create(
-            service_name=config.service_name,
+            service_name=config.serviceName,
             message_service_type=MessagingServiceType.Kafka.name,
-            schema_registry_url=config.schema_registry_url,
-            brokers=config.bootstrap_servers.split(","),
+            config=self.kafka_service_config,
             metadata_config=metadata_config,
         )
         self.schema_registry_client = SchemaRegistryClient(
-            {"url": self.config.schema_registry_url}
+            {"url": self.service_connection.schemaRegistryURL}
         )
         self.admin_client = AdminClient(
```

Comment:
Let's keep it like this for now, although just a heads-up that @ulixius9 was working on preparing a more generic "connection" approach rather than going directly to `Engine` in SQA. This way we can have a generic attribute in all sources with a `get_connection` and dynamically load SQA engines, AWS clients, etc.

Whenever Mayur gets this done, we can come back and have a stab at this

Reply:
missed it before. Its this one https://github.com/open-metadata/OpenMetadata/pull/4316